lib/Runtime/Library/TypedArray.cpp

```diff
@@ -144,10 +144,13 @@ namespace Js
                 if (JavascriptOperators::IsObject(firstArgument))
                 {
                     Var iterator = JavascriptOperators::GetProperty(RecyclableObject::FromVar(firstArgument), PropertyIds::_symbolIterator, scriptContext);
+                    Var arrayIteratorPrototypeNext = nullptr;
 
                     if (!JavascriptOperators::IsUndefinedObject(iterator) &&
                         (iterator != scriptContext->GetLibrary()->GetArrayPrototypeValuesFunction() ||
-                        !JavascriptArray::Is(firstArgument)))
+                            !JavascriptArray::Is(firstArgument) ||
+                            !RuntimeFunction::Is(arrayIteratorPrototypeNext = JavascriptOperators::GetProperty(scriptContext->GetLibrary()->GetArrayIteratorPrototype(), PropertyIds::next, scriptContext)) ||
```

Comment:
Good point.
I can add a flag in JavascriptLibrary called `hasUserDefinedArrayiteratorPrototypeNext` to get around having to call `GetProperty()`.